src/App.tsx

```diff
@@ -1,24 +1,202 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, {
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
 
-const USER_ID = 0;
+import { TodoAppHeader } from './components/TodoAppHeader';
+import { TodoAppFooter } from './components/TodoAppFooter';
+import { ErrorNotification } from './components/ErrorNotification';
+
+import { Todo } from './types/Todo';
+import * as todoService from './api/todos';
+import { FilterLink } from './utils/TodoFilter';
+import { ErrorMessage } from './utils/errorMessages';
+import { TodoList } from './components/TodoList';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedFilter, setSelectedFilter] = useState(FilterLink.All);
+  const [errorMessage, setErrorMessage] = useState(ErrorMessage.Default);
+  const [isRequesting, setIsRequesting] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [processingTodoIds, setProcessingTodoIds] = useState<number[]>([]);
+
+  useEffect(() => {
+    todoService.getTodos(todoService.USER_ID)
+      .then(setTodos)
+      .catch(() => {
+        setErrorMessage(ErrorMessage.Load);
+      });
+  }, []);
+
+  const handleAddTodo = (todoTitle: string): Promise<void> => {
+    setTempTodo({
+      id: 0,
+      userId: todoService.USER_ID,
+      title: todoTitle,
+      completed: false,
+    });
+
+    setIsRequesting(true);
+
+    return todoService
+      .addTodo(todoTitle)
+      .then((newTodo) => {
+        setTodos((prevTodos) => [...prevTodos, newTodo]);
+      })
+      .catch((error) => {
+        setErrorMessage(ErrorMessage.Add);
+        throw error;
```

Comment:
You don't need to return something from the handler and throw an error as you handle it on this step